src/dex/curve-v1/pools/3pool.ts

```diff
@@ -118,7 +118,7 @@ export class ThreePool extends CurvePool {
     const rates = this.getRates();
     const i = _.findIndex(
       this.COINS,
-      c => c.toLowerCase() === this.lastTransferedCoin!.toLowerCase(),
+      c => c.toLowerCase() === this.lastTransferredCoin?.toLowerCase(),
```

Comment:
perhaps you can skip findIndex if lastTransferredCoin is undefined, minor though

Reply:
Yeah, I agree it is better, but I have already published new version :(